src/conn/conn_open.c

```diff
@@ -100,6 +94,10 @@ __wt_connection_close(WT_CONNECTION_IMPL *conn)
 		__wt_yield();
 	}
 
+	/* Shut down the subsystems, ensuring workers see the state change. */
+	F_SET(conn, WT_CONN_CLOSING);
```

Comment:
I believe this was after the subsystem destroy calls so that those subsystems had a chance to do an orderly shutdown of their worker threads. I'm not sure if that's currently required - but I think it's safer to keep the ordering. Why the order change in here? Why is the barrier required? Do some operations need to see the change to `WT_CONN_CLOSING`? The only place I can see that uses this flag to assert anything other than correctness is avoiding having application threads contribute to eviction.